test/helpers/certificate/certificate.go

```diff
@@ -69,7 +68,9 @@ func GenerateSelfSignedCert(t *testing.T, decorators ...SelfSignedCeritificateOp
 		BasicConstraintsValid: true,
 	}
 	derBytes, err := x509.CreateCertificate(rand.Reader, template, template, &privateKey.PublicKey, privateKey)
-	require.NoError(t, err, "failed to create x509 certificate")
+	if err != nil {
+		panic(fmt.Sprintf("failed to create x509 certificate: %s", err))
```

Comment:
I'm not sure I understand why we'd prefer to use panic vs something like `require.NoError()`.

First of all, if we're running more tests it's easier to understand which test failed due to this error (which should rarely happen if at all).

Secondly, if we're running massive amounts of tests - like we do in our current pipeline - we get the failure logs for free if we use something like `require.NoError()`. Why? Imagine that you push a change that would cause a test failure but at the same time cert creation fails. You **would** get the logs from your failed test but the `panic` occurred first so you first need to fix or rerun the test which was caused by the cert failure and then rerun (**again**) the whole suite to then get notified that you test's failed and only then you can figure out what's wrong with it.

Reply:
I would not change it, but there is a problem with usage in `internal/util/test/webhooks.go`.This usage of a global variable is not great, but it allows those variables to be used in `test/integration/suite_test.go` in `func TestMain(m *testing.M)` and other places to ensure that the same certificate is always passed during executing a suite. I don't have an instance of `*testing.T`available to pass. Hence I changed it, because it was pretty easy to do, with not a big cost and resolves described problem. 

For running e.g. 
```
make test.integration.dbless GOTESTFLAGS="-count 1"
```
in case of panic inside of helper, nothing will be run, because globals are evaluated as the first

Reply:
Ok understood. In this case I suggest to kill 2 birds with 1 stone: to alleviate the reused var problem we can introduce a function inside the `github.com/kong/kubernetes-ingress-controller/v2/test/helpers/certificate` package which will return a webhook cert wherever it's needed and it will only be generated once (guarded with something like `sync.Once`).

Instead of getting access to the variable which is shared we can give devs access to this dedicated function which doesn't need to accept `t *testing.T` because of the above mentioned limitations. In order to make this fail fast it will init the variables in `init`. WDYT?

Reply:
I made adjustments. I ensured that certs are generated as fast as possible (hence I decided not to use `sync.Once`, because it would lazily evaluate them when they'd be needed and I wanted to fail as fast as possible in case of an error in cert generation helpers).

Reply:
My bad I missed removing those globals in the previous push (they hadn't been already used).

The implementation from lines `106:114` exactly does it. Kong cert is created once as fast as possible (when the package is imported, before executing inits and `main`), and later the same values are returned by calling `GetKongSystemSelfSignedCerts()` when it is needed. Reading is the only operation, thus data race won't happen. Or am I missing something in my reasoning? 🤔

Reply:
That should do for now. 👍